taichi/codegen/codegen_opengl.cpp

```diff
@@ -153,10 +153,13 @@ class KernelGen : public IRVisitor {
     line_appender_header_.append_raw(kernel_header);
 
     int threads_per_group = opengl_get_threads_per_group();
-    if (num_threads_ < threads_per_group)
-      threads_per_group = std::max(1, num_threads_);
-    else
+    if (num_threads_ <= 0) num_threads_ = 1;
```

Comment:
Out of curiosity, when will `num_threads_` be negative in OpenGL?

Reply:
Maybe `range(6, 4)` or `range(0)` just in case.

Reply:
Ah i see.